js/ui/handlers.js

```diff
@@ -28,6 +28,50 @@ function Handlers(map) {
             map.resize();
             map.update();
         })
+        .on('keydown', function(e) {
+            if (e.altKey || e.ctrlKey || e.metaKey) return;
+            var key = e.keyCode;
+            var keyCodes = {
```

Comment:
could this be distilled into one switch statement rather than the lookup/if structure?